src/Nethermind/Nethermind.Merge.Plugin/EngineRpcModule.Cancun.cs

```diff
@@ -33,16 +33,29 @@ ResultWrapper<PayloadStatusV1> ErrorResult(string error)
                 });
         }
 
-        static IEnumerable<byte[]?> FlattenHashesFromTransactions(ExecutionPayload payload) =>
+        static IEnumerable<byte[]?> FlattenHashesFromTransactions(ExecutionPayloadV3 payload) =>
             payload.GetTransactions()
                 .Where(t => t.BlobVersionedHashes is not null)
                 .SelectMany(t => t.BlobVersionedHashes!);
 
-        return blobVersionedHashes is null ? ErrorResult("Blob versioned hashes must be set")
-            : !FlattenHashesFromTransactions(executionPayload).SequenceEqual(blobVersionedHashes, Bytes.NullableEqualityComparer) ? ErrorResult("Blob versioned hashes do not match")
+        return !FlattenHashesFromTransactions(executionPayload).SequenceEqual(blobVersionedHashes, Bytes.NullableEqualityComparer)
+            ? ErrorResult("Blob versioned hashes do not match")
             : null;
     }
 
+    private ResultWrapper<PayloadStatusV1>? ValidateFork(ExecutionPayload executionPayload)
+    {
+        if (executionPayload.ValidateFork(_specProvider))
+        {
+            return null;
```

Comment:
@flcl42 don't you think that using NULLs to communicate that something is Valid is a bit confusing?

Reply:
It allows to easily chain validations:
`ValidateFork(executionPayload) ?? PreValidatePayload(executionPayload, blobVersionedHashes) ?? NewPayload(executionPayload, 3)`
Do you think it can be named better?